doc/source/cluster/cluster_under_construction/ray-clusters-on-kubernetes/user-guides/gpu.rst

```diff
@@ -0,0 +1,200 @@
+.. _kuberay-gpu:
+
+Using GPUs
+==========
+This document provides tips on GPU usage with Ray on Kubernetes.
+
+To use GPUs on Kubernetes, you will need to configure both your Kubernetes setup and add additional values to your Ray cluster configuration.
+
+To learn about GPU usage on different clouds, see instructions for `GKE`_, for `EKS`_, and for `AKS`_.
+
+The `Ray Docker Hub <https://hub.docker.com/r/rayproject/>`_ hosts CUDA-based images packaged with Ray for use in Kubernetes pods.
+For example, the image ``rayproject/ray-ml:2.0.0-gpu`` is ideal for running GPU-based ML workloads with Ray 2.0.0.
+Read :ref:`here<docker-images>` for further details on Ray images.
+
+Using Nvidia GPUs requires specifying `nvidia.com/gpu` resource `limits` in the container fields of your `RayCluster`'s
+`headGroupSpec` and/or `workerGroupSpecs`.
+(Kubernetes `automatically sets <https://kubernetes.io/docs/tasks/manage-gpus/scheduling-gpus/#using-device-plugins>`_
+the GPU request equal to the limit. However, you might want to specify requests for purposes of documentation.)
+
+Here is a config snippet for a RayCluster workerGroup of up
+to 5 GPU workers.
+
+.. code-block:: yaml
+
+   groupName: gpu-group
+   replicas: 0
+   minReplicas: 0
+   maxReplicas: 5
+   ...
+   template:
+       spec:
+        ...
+        containers:
+         - name: ray-node
+           image: rayproject/ray-ml:2.0.0-gpu
+           ...
+           resources:
+            cpu: 3
+            memory: 50Gi
+            nvidia.com/gpu: 1 # Optional, included just for documentation.
+           limits:
+            cpu: 3
+            memory: 50Gi
+            nvidia.com/gpu: 1 # Required to use GPU.
+            ...
+
+Each of the Ray pods in the group can be scheduled on an AWS `p2.xlarge` instance (1 GPU, 4vCPU, 61Gi RAM).
+
+.. tip::
+
+    GPU instances are expensive -- consider setting up autoscaling for your GPU Ray workers,
+    as demonstrated with the `minReplicas:0` and `maxReplicas:5` settings above.
+    To enable autoscaling, remember also to set `enableInTreeAutoscaling:True` in your RayCluster's `spec`
+    Finally, make sure your group or pool of GPU Kubernetes nodes are configured to autoscale.
+
+GPUs and Ray
+____________
+
+The KubeRay operator advertises container GPU resource limits to
+the Ray scheduler and the Ray autoscaler. In particular, the Ray container's
+`ray start` entrypoint will be automatically configured with the appropriate `--num-gpus` option.
+
+* Learn more about Ray's :ref:`gpu-support`.
+
+GPU workload scheduling
+~~~~~~~~~~~~~~~~~~~~~~~
+After a Ray pod with access to GPU is deployed, it will
```

Comment:
Can we also add some info about what is required (if anything) to make the ML libraries use GPUs? E.g. TensorFlow or Pytorch. Maybe link to the section in the documentation that talks about this (I hope there is one).

Reply:
(This is probably what most people will be interested in doing)

Reply:
I think the `rayproject/ray-ml:gpu` are (or should be) packaged with everything required for the Ray ML libraries covered in the Ray docs.

I will make this assertion.
@richardliaw does that sound reasonable?

Reply:
Added a bit of exposition.